### PR TITLE
Fix dimensionality bug in GenerateXdmf script

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Quick fix for `GenerateXdmf.py` script to incorporate 2D volume files.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Visualization/Python/GenerateXdmf.py [hunk 1]
```diff
@@ -214,7 +214,13 @@ def data_item(data_type):
                         "AttributeType=\"Vector\" Center=\"Node\">\n" %
                         (vector))
                     xdmf_output += data_item_vec
-                    for index in ["_x", "_y", "_z"]:
+
+                    index_list = []
+                    if dimensionality == 2:
+                        index_list = ["_x", "_y"]
+                    elif dimensionality == 3:
+                        index_list = ["_x", "_y", "_z"]
```
Comment (fmahebert) [inline src/Visualization/Python/GenerateXdmf.py]: This looks good, but what do you think of writing it in fewer lines like this?
```
index_list = ["_x", "_y"]
if dimensionality == 3:
    index_list.append("_z")
```
(I haven't tested this works, but something like this should...)

Comment (yoonso0-0) [reply src/Visualization/Python/GenerateXdmf.py]: @fmahebert  Worked out successfully (at least for advection 2D data) -- do you have any 3D volume data in mind for a quick double-check test?

Comment (fmahebert) [reply src/Visualization/Python/GenerateXdmf.py]: I think the NewtonianEuler 3D Riemann problem should be fairly easy to use as a check: I think you can add some options to the input file to observe the density (a scalar) and the velocity (a vector) in 3D and that should be a sufficient check.

Comment (yoonso0-0) [reply src/Visualization/Python/GenerateXdmf.py]: Check completed

**Review by nilsvu — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**